core/src/main/java/net/opentsdb/query/ReadCacheQueryPipelineContext.java

```diff
@@ -1329,6 +1347,81 @@ public int compareTo(Object o) {
     return ids;
   }
   
+  //Returns stuff like <nodeID>:<dataSourceNodeID>
+  public List<String> getDataSourceIds(final MutableGraph<QueryNodeConfig> config_graph, final QueryNodeConfig node) {
+    if (node instanceof MergerConfig) {
+      if (!node.getId().equals(((MergerConfig) node).getDataSource())) {
+        return Lists.newArrayList(node.getId() + ":" 
+            + ((MergerConfig) node).getDataSource());
+      }
+      return Lists.newArrayList(((MergerConfig) node).getDataSource() + ":" 
+          + ((MergerConfig) node).getDataSource());
+    } else if (node.joins()) {
+      return Lists.newArrayList(node.getId() + ":" + node.getId());
+    } else if (node instanceof TimeSeriesDataSourceConfig) {
+      return Lists.newArrayList(node.getId() + ":" 
+          + ((TimeSeriesDataSourceConfig) node).getDataSourceId());
+    }
+    
+    List<String> sources = null;
```

Comment:
the variable `sources` is only used inside the `if` block at line number: 1369. Move it inside the `if` block please